bot/exts/help_channels/_cog.py

```diff
@@ -439,8 +436,9 @@ async def on_message(self, message: discord.Message) -> None:
         if not is_available or _channel.is_excluded_channel(channel):
             return  # Ignore messages outside the Available category or in excluded channels.
 
-        log.trace("Waiting for the cog to be ready before processing messages.")
-        await self.ready.wait()
+        if not self.init_task.done():
+            log.trace("Waiting for the cog to be ready before processing messages.")
+            await self.init_task
```

Comment:
I suppose you don't even need the `.done()` check first, although it makes sense from a logical point of view. Not sure about the performance characteristics, but I doubt it really matters here anyway.

Reply:
I think I did this to avoid sending the log message, but I now realise it causes a bug so I've removed the `done()` check. See the message of d9f87cc for more details.